router/router.go

```diff
@@ -725,44 +728,102 @@ func (worker *workerT) handleWorkerDestinationJobs(ctx context.Context) {
+			_destinationJobMetadata := destinationJobMetadata
 
-			status := jobsdb.JobStatusT{
-				JobID:         destinationJobMetadata.JobID,
-				ExecTime:      time.Now(),
-				RetryTime:     time.Now(),
-				AttemptNum:    attemptNum,
-				ErrorCode:     strconv.Itoa(respStatusCode),
-				ErrorResponse: []byte(`{}`),
-				Parameters:    []byte(`{}`),
+			routerJobResponses = append(routerJobResponses, &RouterJobResponse{
+				jobID:                  destinationJobMetadata.JobID,
+				destinationJob:         &_destinationJob,
+				destinationJobMetadata: &_destinationJobMetadata,
+				respStatusCode:         respStatusCode,
+				respBody:               respBody,
+				attemptedToSendTheJob:  attemptedToSendTheJob,
+			})
+		}
+	}
+
+	sort.Slice(routerJobResponses, func(i, j int) bool {
+		return routerJobResponses[i].jobID < routerJobResponses[j].jobID
+	})
+
+	destLiveEventSentMap := make(map[*types.DestinationJobT]struct{})
+	//Struct to hold unique users in the batch (worker.destinationJobs)
+	userToJobIDMap := make(map[string]int64)
+
+	for _, routerJobResponse := range routerJobResponses {
+		destinationJobMetadata := routerJobResponse.destinationJobMetadata
+		destinationJob := routerJobResponse.destinationJob
+		attemptedToSendTheJob := routerJobResponse.attemptedToSendTheJob
+		attemptNum := destinationJobMetadata.AttemptNum
+		if attemptedToSendTheJob {
+			attemptNum++
+		}
+		status := jobsdb.JobStatusT{
+			JobID:      destinationJobMetadata.JobID,
+			AttemptNum: attemptNum,
+			ExecTime:   time.Now(),
+			RetryTime:  time.Now(),
+			Parameters: []byte(`{}`),
+		}
+		if !isSuccessStatus(respStatusCode) {
+			if prevFailedJobID, ok := userToJobIDMap[destinationJobMetadata.UserID]; ok {
+				//This means more than two jobs of the same user are in the batch & the batch job is failed
+				//Only one job is marked failed and the rest are marked waiting
+				//Job order logic requires that at any point of time, we should have only one failed job per user
+				//This is introduced to ensure the above statement
+				resp := fmt.Sprintf(`{"blocking_id":"%d", "user_id":"%s", "moreinfo": "attempted to send in a batch"}`, prevFailedJobID, destinationJobMetadata.UserID)
+				status.JobState = jobsdb.Waiting.State
+				status.ErrorResponse = []byte(resp)
+				routerJobResponse.status = status
+				worker.rt.responseQ <- jobResponseT{status: &status, worker: worker, userID: destinationJobMetadata.UserID, JobT: destinationJobMetadata.JobT}
+				continue
+			} else {
+				userToJobIDMap[destinationJobMetadata.UserID] = destinationJobMetadata.JobID
 			}
+		}
 
-			worker.postStatusOnResponseQ(respStatusCode, respBody, destinationJob.Message, respContentType, &destinationJobMetadata, &status)
+		status.ErrorResponse = []byte(`{}`)
+		status.ErrorCode = strconv.Itoa(respStatusCode)
+		if isSuccessStatus(respStatusCode) {
+			status.JobState = jobsdb.Succeeded.State
+		} else {
+			status.JobState = jobsdb.Failed.State
```

Comment:
Is this appropriate..?
Added this because the live-event endopint needs a `jobState`, otherwise it's a 400(invalid event).
Could be failed/aborted, thought of going with failed here.